src/frontend/src/server.rs

```diff
@@ -69,8 +74,12 @@ impl Services {
                     .context(error::RuntimeResourceSnafu)?,
             );
 
-            let mysql_server =
-                MysqlServer::create_server(instance.clone(), mysql_io_runtime, opts.tls.clone());
+            let mysql_server = MysqlServer::create_server(
+                instance.clone(),
+                mysql_io_runtime,
+                opts.tls.clone(),
+                user_provider.clone(),
```

Comment:
Consider to add a `ProviderRegistry` to contain all feature providers and pass it into mysql/pg servers. We are going to have some providers like this in near future.